### PR TITLE
Refactor trackItemDragHandler - PMT #109430

### PR DESCRIPTION
**Pull request by nikolas**
* This addresses the problem where the dirty state is set to true when
  you only click a track element.
* In the jest tests I had `.toBeNull` everywhere, but I found out this
  doesn't fail when it should. I changed them all to `.toBeNull()`.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -383,3 +383,44 @@ export function reassignKeys(track) {
...
+    let newTrack = origTrack.slice();
+    const elem = _.find(newTrack, ['key', parseInt(item['i'], 10)]);
+    const percent = item.x / 1000;
+    const startTime = percent * sequenceDuration;
+
+    // Check that the value actually needs to be updated.
+    if (startTime === origElem.start_time ||
+        // Allow just over a hundredth of a second margin of
+        // error. This can occur when clicking the track elements.
+        Math.abs(startTime - origElem.start_time) <= 0.015
```
Comment (nikolas) [inline src/utils.js]: I just realized this 0.015 value is dependent on the sequenceDuration - this should be larger for long videos and smaller for short ones.

Comment (sdreher) [inline src/utils.js]: Yeah, this works reliably with a 30-second vid, but not so well with a 6-minute vid. unless I'm really, really careful when I click the item. :-)

Comment (nikolas) [inline src/utils.js]: I've updated this and am testing with spine videos of various lengths.

Comment (sdreher) [inline src/utils.js]: definitely an improvement.

**Pull request merged**